### PR TITLE
Add infinite perspective matrix

### PR DESCRIPTION
**Pull request by d07RiV**
Support for zFar = infinity for projection matrix.

Dunno why dist build ended up so different.

### CODE REVIEW

**Comment by stefnotch** [comment]
@d07RiV Wouldn't it be cleaner to simply modify https://github.com/toji/gl-matrix/blob/f1871a7c4495d441573258bfdf995774da08797f/src/gl-matrix/mat4.js#L1268 to support setting the far value to `Math.Infinity`?

**Comment by d07RiV** [comment]
That seems less intuitive to me, but if you think that's better then maybe.

**Comment by stefnotch** [comment]
@d07RiV Well, I'd prefer to not add a ton of functions that might not be used as often.
`less intuitive`: Well, other than adding another function, how could you make it more intuitive?

**Comment by d07RiV** [comment]
You think null value for `far` is better?

**Comment by stefnotch** [comment]
@d07RiV That seems equally good/bad. I'd say, let's just go with Math.Infinity and add a note in the documentation. Something like `@param {number} far Far bound of the frustum, supports Math.Infinity`

File: src/gl-matrix/mat4.js [hunk 1]
```diff
@@ -1278,42 +1278,18 @@ export function perspective(out, fovy, aspect, near, far) {
...
-  out[7] = 0;
-  out[8] = 0;
-  out[9] = 0;
-  out[10] = -1;
-  out[11] = -1;
-  out[12] = 0;
-  out[13] = 0;
-  out[14] = -2 * near;
   out[15] = 0;
+  if (far != null && far !== Infinity) {
```
Comment (stefnotch) [inline src/gl-matrix/mat4.js]: That's also fine. 🙂 If anyone has an issue with it, they can just open an issue.

**Comment by d07RiV** [comment]
Why not both?

**Comment by d07RiV** [comment]
Also I'm not sure why build was changed so much, and I didn't generate docs because it would put my time/timezone in every file.

**Comment by stefnotch** [comment]
@d07RiV Well, the checks passed and the PR adds some useful functionality, so, merging this!
Thank you! 👍

**Pull request merged**